### ISSUE TITLE
CUDA onnxruntime-gpu=1.12.1 inference result differs from onnxruntime-gpu=1.11.0

### ISSUE THREAD

**Issue by iperov** [microsoft/onnxruntime#12706]
**Describe the bug**

1.12.1 inference result differs from 1.11.0
The bug only with CUDA.
Works fine on CPU and DML.

**Urgency**
none

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): win 10
- ONNX Runtime installed from (source or binary): pip
- ONNX Runtime version: 1.12.1
- Python version: 3.7
- CUDA/cuDNN version: 11.5
- GPU model and memory: RTX 3090

**To Reproduce**

dl https://github.com/iperov/DeepFaceLive/releases/download/trash/onnxruntime-1-12-1-bug.zip
unzip
run repro.py with onnxruntime-gpu==1.11.0 you will see the pic
![onnxruntime1 11 0](https://user-images.githubusercontent.com/8076202/186488732-38d07921-255a-4c47-ada4-a36d5a07b1b7.png)

run repro.py with onnxruntime-gpu==1.12.1 you will see different pic
![onnxruntime 1 12 1](https://user-images.githubusercontent.com/8076202/186488739-0ed6772b-0191-4944-b798-5355b32d7c0b.png)

**Expected behavior**
Expected same inference result as on 1.11.0

**Comment by pranavsharma** [comment]
To rule out any cuda related deps, can you ensure you're using the exact same cuda deps in both cases?

**Comment by iperov** [comment]
yes. I am using isolated environment, and just change onnxruntime version via pip.

**Comment by iperov** [comment]
generator.onnx is very big, and I have no idea what OP produces that glitch.

**Comment by pranavsharma** [comment]
One way to find out the glitch is to build ORT with  --cmake_extra_defines onnxruntime_DEBUG_NODE_INPUTS_OUTPUTS=ON and then compare the node outputs side by side.

**Comment by iperov** [comment]
Good luck.

**Comment by xuguozhi** [comment]
I have met the same issue when deepfaking

**Comment by hariharans29** [comment]
@iperov - I scanned the ops that have had some (even minor) changes since 1.11.0 - there are 3-4 and all show up in the model. So, I ll try and dig more into this. Another point of failure could be changes in graph optimizations that introduced new bugs. A quick way to rule that out will be to see if setting Graph Opt Level to ORT_ENABLE_BASIC and checking to see if the content diff persists.

**Comment by hariharans29** [comment]
> I have met the same issue when deepfaking

@xuguozhi - Please open a separate issue with your model and more details.

**Comment by iperov** [comment]
@hariharans29 

only works fine with `rt.GraphOptimizationLevel.ORT_DISABLE_ALL`

**Comment by hariharans29** [comment]
> @hariharans29
> 
> only works fine with `rt.GraphOptimizationLevel.ORT_DISABLE_ALL`

Thanks @iperov . That is a good data point. That means it is a graph optimization issue of some sort (may not be related to the ops themselves). I presume in 1.11.0, it worked fine even with default graph optimization (i.e.) ORT_ENABLE_ALL ?

**Comment by iperov** [comment]
yes

**Comment by hariharans29** [comment]
This should fixed in master now. If there is a patch release for 1.12, I ll make sure this makes it there.

**Comment by iperov** [comment]
@hariharans29 thank you very much
